### PR TITLE
Support for bind mount of fuse mounts.

### PR DESCRIPTION
**Pull request by talakad**
Bind mount of fuse mounts show up with fuse name  as device rather than device path being bind mounted. Added mechanism to pass in device name for fuse mounts. 

To elaborate, Mount function is passed with a device parameter and a path to mount the device with. In case of bind mount, both device and mount path are simply paths. After the bind mount, both of them will show with filesystem/device  that is on source path.  So mounter needs to track the mounts with this filesystem "string" passed in the options for bind mounts to reflect the correct mount status.

### CODE REVIEW

File: pkg/mount/mount.go [hunk 1]
```diff
@@ -323,12 +322,17 @@ func (m *Mounter) deletePath(path string) bool {
...
 func (m *Mounter) Mount(
 	minor int,
-	device, path, fs string,
+	devPath, path, fs string,
 	flags uintptr,
 	data string,
 	timeout int,
 	opts map[string]string,
 ) error {
+	device := devPath
```
Comment (lpabon) [inline pkg/mount/mount.go]: Could you add a comment here that device is initialized to devPath, but could be overwritten by opts[]?

File: pkg/mount/mount.go [hunk 2]
```diff
@@ -323,12 +322,20 @@ func (m *Mounter) deletePath(path string) bool {
...
 	flags uintptr,
 	data string,
 	timeout int,
 	opts map[string]string,
 ) error {
+	// device gets overwritten if opts specifies fuse mount with
+	// options.OptionsDeviceFuseMount.
+	device := devPath
+	if value, ok := opts[options.OptionsDeviceFuseMount]; ok {
+		// fuse mounts show-up with with this key as device.
```
Comment (lpabon) [inline pkg/mount/mount.go]: 'with with'

File: pkg/options/options.go [hunk 3]
```diff
@@ -17,6 +17,8 @@ const (
 	OptionsWaitBeforeDelete = "WAIT_BEFORE_DELETE"
 	// OptionsRedirectDetach Redirect detach to the node where volume is attached
 	OptionsRedirectDetach = "REDIRECT_DETACH"
+	// OptionsDeviceFuseMount name of fuse mount device
+	OptionsDeviceFuseMount = "DEV_FUSE_MOUNT"
```
Comment (disrani-px) [inline pkg/options/options.go]: Fuse for shared volumes is a px specific implementation. libopenstorage doesn't know about it, right? So should this be called DeviceVirtualMount (or something similar) instead?

File: pkg/mount/mount.go [hunk 4]
```diff
@@ -8,6 +8,10 @@ import (
 	"encoding/hex"
 	"errors"
 	"fmt"
+	"github.com/libopenstorage/openstorage/pkg/keylock"
```
Comment (disrani-px) [inline pkg/mount/mount.go]: This was fine before, separated from the std imports

**Pull request merged**